cmd/fs/helpers.go

```diff
@@ -46,6 +48,65 @@ func filerForPath(ctx context.Context, fullPath string) (filer.Filer, string, er
 	return f, path, err
 }
 
+const dbfsPrefix string = "dbfs:/"
+
 func isDbfsPath(path string) bool {
-	return strings.HasPrefix(path, "dbfs:/")
+	return strings.HasPrefix(path, dbfsPrefix)
+}
+
+func getValidArgsFunction(pathArgCount int, onlyDirs bool, filerForPathFunc func(ctx context.Context, fullPath string) (filer.Filer, string, error)) func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+	return func(cmd *cobra.Command, args []string, toComplete string) ([]string, cobra.ShellCompDirective) {
+		cmd.SetContext(root.SkipPrompt(cmd.Context()))
+
+		err := mustWorkspaceClient(cmd, args) // TODO: Fix this
+		if err != nil {
+			return nil, cobra.ShellCompDirectiveError
+		}
+
+		isValidPrefix := isDbfsPath(toComplete)
+
+		if !isValidPrefix {
+			return []string{dbfsPrefix}, cobra.ShellCompDirectiveNoSpace
+		}
+
+		filer, path, err := filerForPathFunc(cmd.Context(), toComplete)
```

Comment:
you could store `ctx := cmd.Context()` and use it everywhere in the code

Reply:
Good point, fixed! The repeated Context() calling came from here: https://github.com/databricks/cli/blob/e6241e196fe2ae37bcf6709243eabdeeb94f261a/cmd/sync/sync.go#L146

Reply:
Oh, I see why they're calling `cmd.Context()` each time in sync.go:`root.MustWorkspaceClient` modifies the context on the command.